### ISSUE TITLE
TSLint fails to compile with Typescript 2.6

### ISSUE THREAD

**Issue by sandersn** [palantir/tslint#3396]
### Bug Report

- __TSLint version__: master
- __TypeScript version__: typescript@next (failures are the same with 2.6, though)
- __Running TSLint via__: not running, just compiling

#### Actual behavior
Two compile errors:
1. [tslib needs to be updated](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript) for the new emit helper `__makeTemplateObject`, used for template literals.
2. 2.6's improved checking of computed property names in object literals found a typo in a constant in completedDocsRule.ts.

```ts
export const TAGS_FOR_EXISTENCE = "exists";
// should be
export const TAGS_FOR_EXISTENCE = "existence";
```

#### Expected behavior
No compile errors.

**Comment by ajafff** [comment]
@sandersn I already have a PR open to address these problems

**Comment by sandersn** [comment]
#3320 differs in two main ways:

1. Doesn't address the failure in completedDocsRule.
2. Upgrades to Typescript 2.6.

#3397 can be merged ahead of the actual upgrade to 2.6. I'm not sure which is the better process; you can easily add completedDocsRule fix if you want, and I'll close #3397.

**Comment by ajafff** [comment]
@sanders You're right, it makes sense to fix this now. I thought you were about to open a PR to upgrade to 2.6.
Thanks for fixing.
